File: 3014-minimum-number-of-pushes-to-type-word-i/3014-minimum-number-of-pushes-to-type-word-i.go
func minimumPushes(word string) int {
    ans := 0
    letters := make([]int, 26)
    count := 0
    for i := 0; i < len(word); i++ {
        c := word[i] - 'a'
        if letters[c] == 0 {
            letters[c] = (count / 8) + 1
            ans += letters[c]
            count++
        } else {
            ans += letters[c]
        }
    }

    return ans
}

// greedy, math, string
// time: O(n)
// space: O(1)

File: 3014-minimum-number-of-pushes-to-type-word-i/3014-minimum-number-of-pushes-to-type-word-i.py
class Solution:
    def minimumPushes(self, word: str) -> int:
        ans = 0
        letters = [0] * 26
        count = 0
        for i in range(len(word)):
            c = ord(word[i]) - ord('a')
            if not letters[c]:
                letters[c] = (count // 8) + 1
                ans += letters[c]
                count += 1
            else:
                ans += letters[c]
        
        return ans

# greedy, math, string
# time: O(n)
# space: O(1)